SIL.Core.Desktop/Progress/Commands/BasicCommand.cs

```diff
@@ -51,24 +50,22 @@ public StatusCallback SecondaryStatusTextCallback
 		/// <summary>
 		/// Implementation of the async work invoker
 		/// </summary>
-		/// <remarks>
-		/// We're using the delegate BeginInvoke() / EndInvoke() pattern here
-		/// </remarks>
-		protected override void BeginInvokeCore()
+		protected override async Task BeginInvokeCore()
 		{
-			WorkInvoker worker = new WorkInvoker( DoWork );
-			worker.BeginInvoke(
+			WorkInvoker worker = DoWork;
+			var workTask = Task.Run(() => worker.Invoke(
 				_initializeCallback,
-				_progressCallback ,
-				_primaryStatusTextCallback ,
-				_secondaryStatusTextCallback,
-				new AsyncCallback( EndWork ), null );
+				_progressCallback,
+				_primaryStatusTextCallback,
+				_secondaryStatusTextCallback));
+			await EndWork(workTask).ConfigureAwait(false);
 		}
 
-		protected override void BeginInvokeCore2(ProgressState progress)
+		protected override async Task BeginInvokeCore2(ProgressState progress)
 		{
-			WorkInvoker2 worker = new WorkInvoker2(DoWork2);
-			worker.BeginInvoke(progress, new AsyncCallback(EndWork2), null);
+			WorkInvoker2 worker = DoWork2;
+			var workTask = Task.Run(() => worker.Invoke(progress));
+			await EndWork(workTask).ConfigureAwait(false);
```

Comment:
Can you add a comment why `ConfigureAwait` is necessary? I notice you have that information in the commit message, but IMO it would be good to have it in the code as well.

Reply:
It's not necessary, but it makes the code more compatible. By default, an await captures the context of the method and requires that context to resume once the Task finishes. This _can_ cause deadlock issues in some situations, but .ConfigureAwait(false) indicates that the resumption can occur on any thread. This can apply here, since no information from the Task is required to resume the method.

Reply:
If you want me to put that in the code, I can, it's just kind of a brittle comment